lib/internal/source_map.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const _sourceMapCache = {};
+function maybeStoreSourceMap(filename, content) {
+  if (!process.env.NODE_V8_COVERAGE) return;
+  const match = content.match(/\/[*/]#\s+sourceMappingURL=(?<sourceMap>[^\s]+)/);
+  if (match) {
+    _sourceMapCache[filename] = match.groups.sourceMap;
+  }
+}
+
+function writeSourceMapsOnExit() {
+  const reallyReallyExit = process.reallyExit;
+
+  process.reallyExit = function(code) {
+    writeSourceMaps();
+    reallyReallyExit(code);
+  };
+
+  process.on('exit', writeSourceMaps);
+}
+
+function writeSourceMaps() {
+  const { join } = require('path');
```

Comment:
If possible I'd prefer us do this together with other V8 profiles that we write, so there are fewer monkey-patchable exit hooks scattered around in the code base - see https://github.com/nodejs/node/blob/e3f4ec94b8360414ecff65d8c04d9fa15d510c7f/src/inspector_profiler.cc#L301-L320

Reply:
I'm not convinced that the cache of source maps should be something that we delegate to the inspector. I'm expecting that we will likely maintain our own cache in Node.js.

Is there an existing mechanism we could piggyback off of that would allow us to write a JavaScript object to a cache directory at exit time?

Reply:
> Is there an existing mechanism we could piggyback off of that would allow us to write a JavaScript object to a cache directory at exit time?

This is the place where we "write a JavaScript object to a file in a directory at exit time". The function is named `EndStartedProfilers()` but what it does does not require an inspector session.